### PR TITLE
fix: prevent saving 555 assessments (#189)

### LINKED ISSUE #189: Prevent saving daily assesment results for 555 numbers
This will allow us to insert 555 test accounts which will be non-mutable and use them both for e2e and manual testing.

### PR DESCRIPTION
**Pull request by fparga**
## Description

Prevent saving assessments and cancelling reminders on test accounts.

## Related issues

* closes #189 

## Checklist

- [X] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [X] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: action-server/covidflow/utils/persistence.py [hunk 1]
```diff
@@ -1,89 +1,115 @@
...
+logger = get_logger(__name__)
 
+PHONE_NUMBER_SLOT = "phone_number"
 METADATA_SLOT = "metadata"
 REMINDER_ID_METADATA_PROPERTY = "reminder_id"
 
 
-def get_reminder_id(slot_values: Dict[Text, Any]) -> int:
-    metadata = slot_values.get(METADATA_SLOT, {})
+def _get_reminder_id(slot_values: Dict[Text, Any]) -> int:
```
Comment (NetBKarine) [inline action-server/covidflow/utils/persistence.py:19]: You have private functions defined at the top and the bottom of the file. We are not coherent between files, but in one file they should all be at the same place

File: action-server/tests/actions/test_daily_ci_enroll_form.py [hunk 2]
```diff
@@ -619,7 +632,8 @@ def test_provide_preconditions_explain(self):
         with self.assertRaises(ActionExecutionRejection):
             self.run_form(tracker)
 
-    def test_provide_has_dialogue_affirm(self):
+    @patch("covidflow.actions.daily_ci_enroll_form.ci_enroll", return_value=True)
```
Comment (NetBKarine) [inline action-server/tests/actions/test_daily_ci_enroll_form.py:635]: The real thing doesn't return anything now, the mock should return None.

File: action-server/tests/actions/test_daily_ci_enroll_form.py [hunk 3]
```diff
@@ -650,7 +664,8 @@ def test_provide_has_dialogue_affirm(self):
             ]
         )
 
-    def test_provide_has_dialogue_deny(self):
+    @patch("covidflow.actions.daily_ci_enroll_form.ci_enroll", return_value=True)
```
Comment (NetBKarine) [inline action-server/tests/actions/test_daily_ci_enroll_form.py:667]: Idem here

File: action-server/covidflow/utils/persistence.py [hunk 4]
```diff
@@ -1,89 +1,115 @@
...
+    logger.debug("Saving assessment")
 
     session = session_factory()
     try:
-        session.add(Assessment.create_from_slot_values(reminder_id, slot_values))
+        if _is_test_session(session, slot_values):
+            return
+
+        reminder_id = reminder_id or _get_reminder_id(slot_values)
+        assessment = Assessment.create_from_slot_values(reminder_id, slot_values)
```
Comment (NetBKarine) [inline action-server/covidflow/utils/persistence.py:93]: You should call _save_assessment here

File: action-server/covidflow/utils/persistence.py [hunk 5]
```diff
@@ -1,89 +1,115 @@
...
+        session.add(assessment)
+
         session.commit()
+
+        logger.debug("Saved assessment", assessment=assessment)
     except:
-        logger.exception("Could not save assessment")
         session.rollback()
+        raise
```
Comment (NetBKarine) [inline action-server/covidflow/utils/persistence.py:101]: If you make it raise an error here, then you should catch it in daily_ci_assessment_common and do nothing with it. If we can't save one day's assessment, it should not disrupt the dialogue. Raising an error will stop the action server, the form will fail, and the dialogue will do weird things. We don't want that.

File: action-server/tests/actions/test_daily_ci_enroll_form.py (hunk 2 above)
Comment (fparga) [reply action-server/tests/actions/test_daily_ci_enroll_form.py:635]: Indeed, leftover from before

File: action-server/covidflow/utils/persistence.py (hunk 4 above)
Comment (fparga) [reply action-server/covidflow/utils/persistence.py:93]: Right

File: action-server/covidflow/utils/persistence.py (hunk 5 above)
Comment (fparga) [reply action-server/covidflow/utils/persistence.py:101]: Good point, I missed that one. Will also add try except for `cancel_reminder`

File: action-server/covidflow/utils/persistence.py [hunk 6]
```diff
@@ -1,89 +1,115 @@
...
 
-def get_reminder_id(slot_values: Dict[Text, Any]) -> int:
-    metadata = slot_values.get(METADATA_SLOT, {})
+def _get_reminder_id(slot_values: Dict[Text, Any]) -> int:
+    try:
+        metadata = slot_values[METADATA_SLOT]
+        hashed_id = metadata[REMINDER_ID_METADATA_PROPERTY]
+        return decode_reminder_id(hashed_id)
+    except:
+        raise
```
[the remark this answers is not in the document]
Comment (fparga) [reply action-server/covidflow/utils/persistence.py]: No use at all

File: action-server/covidflow/utils/persistence.py (hunk 1 above)
Comment (fparga) [reply action-server/covidflow/utils/persistence.py]: Moved everything to the bottom

**Pull request merged**